python_modules/libraries/dagster-aws/dagster_aws/emr/emr_eks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,569 @@
+      add it to the Python path, and use that to import modules.
+
+      **Note**: Python implicit namespace packages are not supported.
+      All packages **must** have an `__init__.py` to be importable
+      in this workflow.
+
+      **Note**: if the container used has some code, it will just be
+      ignored. For instance, we can run with a container that has the code
+      from `develop` or a previous version of the code from the branch. As
+      we are staging the local code, code on the container will be ignored.
+
+    The container-only workflow is selected automatically when the Dagster
+    repository is seen to come from a docker image. In other cases, the
+    code-upload workflow is used.
+    """
+
+    def __init__(
+        self,
+        cluster_id,
+        container_image,
+        deploy_local_job_package,
+        emr_release_label,
+        job_role_arn,
+        log_group_name,
+        log4j_conf,
+        region_name,
+        staging_bucket,
+        staging_prefix,
+        spark_config,
+        ephemeral_instance,
+        additional_relative_paths,
+        working_directory_override,
+    ):
+        EmrPySparkStepLauncherBase.__init__(
+            self,
+            region_name,
+            staging_bucket,
+            staging_prefix,
+            deploy_local_job_package,
+            cluster_id,
+            spark_config,
+            emr_eks_step_main.__file__,
+        )
+
+        self.container_image = check.str_param(container_image, "container_image")
+        self.emr_release_label = check.str_param(emr_release_label, "emr_release_label")
+        self.job_role_arn = check.str_param(job_role_arn, "job_role_arn")
+        self.log_group_name = check.str_param(log_group_name, "log_group_name")
+        self.additional_relative_paths = additional_relative_paths
+        self.log4j_conf = log4j_conf
+        self.working_directory_override = working_directory_override
+
+        # TODO: all this configuration business is extremely verbose. How can we
+        # make it more streamlined? Each parameter is mentioned 4 times, and the
+        # types are mentioned twices here and above.
+
+        if ephemeral_instance:
+            # The step launcher resource is first created in
+            # the plan process, from where a job run is created.
+            #
+            # The resource is then created again when we run the
+            # step in the Spark driver on the EMR cluster.
+            #
+            # A Spark session is only required in the latter case,
+            # so we only create it on ephemeral Dagster instances
+            # to avoid wasting time initializing a Spark session
+            # that will not be used in the plan process.
+            #
+            # We do not pass any configuration to the session, as
+            # it will be defined when we submit the job run instead.
+            super(EmrEksPySparkResource, self).__init__({})
```

Comment:
this is specific to the fact that our step launcher is also a PySpark resource, I wonder what to do with it...